### PR TITLE
Memory leak cleanup : signals manager

### PR DESCRIPTION
**Pull request by havietisov**
### Purpose
Adds changes necessary in signals manager to enable future cleanups

### Notes:
I don't expect anything to break, but please check.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Managers/SignalsManager.cs [hunk 1]
```diff
@@ -17,7 +17,30 @@ public class SignalsManager : SingletonManager<SignalsManager>
...
+			Receivers.Clear();
+		}
+
+		public override void OnDestroy()
+		{
+			EventManager.RemoveHandler(Event.SceneUnloading, ClearReceivers);
+			base.OnDestroy();
+		}
+
+		public void Clear()
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Managers/SignalsManager.cs:33]: Will it be a separate PR hooking all of these up?

**Review by Bod9001 — APPROVED** [review]

**Review by corp-0 — APPROVED** [review]
